bgpd/bgpd.h

```diff
@@ -1213,8 +1213,16 @@ struct addpath_paths_limit {
 	uint16_t receive;
 };
 
+enum connection_direction {
+	UNKNOWN,
+	CONNECTION_INCOMING,
+	CONNECTION_OUTGOING,
+	ESTABLISHED,
```

Comment:
established and direction does not sound very common or how do we need to interpret it?

Reply:
So BGP in the starting stages of the FSM, it is possible that both this BGP and the remote BGP processes are opening a connection at the same time.  The CONNECTION_INCOMING is to signify the `struct peer_connection` that is being used to create an incoming connection.  The CONNECTION_OUTGOING is to signify the 'struct peer_connection' that is being used to create an outgoing connection.  At some point the state machine decides that one of the two connections is going to be used, the one that is being thrown away gets closed off and the one being kept gets marked as ESTABLISHED.  I don't have a better name.  Do you have a suggestion?

Reply:
I added a big statement to the code as to what I am trying to do